### ISSUE TITLE
Runtime error on compiling todomvc

### ISSUE THREAD

**Issue by Hazelfire** [tweag/asterius#389]
Hey! This is a fantastic project! Thank you so much for all the work in developing the future of web.

On the current head, I'm having a  bit of trouble compiling simple projects. I get the following error inside the firefox console, the webpage loads but no functionality works:
```
RuntimeError: @http://localhost:8080/mayfront.wasm:wasm-function[4676]:0x1ca357
scheduler_loop@http://localhost:8080/rts.scheduler.mjs:424:15
rts.scheduler.mjs:298:15

uncaught exception: Scheduler died with: @http://localhost:8080/mayfront.wasm:wasm-function[4676]:0x1ca357
scheduler_loop@http://localhost:8080/rts.scheduler.mjs:424:15
```

I don't use the docker image, and instead compile the head myself in an Arch Linux environment.

Actually printing the error (at rts.scheduler.mjs:291) gives: 

```
RuntimeError: "invalid conversion to integer"
    scheduler_loop http://localhost:8080/rts.scheduler.mjs:425
```
Any ideas on what causes this? I'll do some further investigation to the source of this error. I'm new to this project, forgive me if I've missed something. I'll triple check that this is replicating. I'm currently using ahc-link on the todomvc directory.

```
ahc-link --input-hs todomvc/todomvc.hs --output-directory build --browser
```

**Comment by TerrorJack** [comment]
Hi, thanks for reporting! We'll manually check this with an arch lin ux container later. It's also possible that the `todomvc` example itself bitrotted since we don't have headless testing on CI yet.

**Comment by Hazelfire** [comment]
Hey! Thanks for the response. I'm very sorry for not providing a replicating test, I was a little short for time.

This configuration seems to do it for me:

https://github.com/Hazelfire/replication-test

I have a strange feeling that it could be the latter... it already is a little bit rotted as the index.html file wants to point to a `todomvc.js` when what is emmitted in `todomvc.mjs`

I couldn't get the issue to replicate with a much simpler "Hello World" program. I'll soon add that to the repo too for reference.

I would be very interested in helping get headless testing to work if it doesn't already, it's currenly my mission to try and get `cabal new-test` to work in my own project.

Also, I do have a Vagrantfile for arch linux which compiles ahc if you're interested (or really, currently copies over the binaries and bootstrap that I've already pre-compiled, because asterius does take a while to compile). I'll first check if your debian docker image compiled on the current head replicates before doing that. But if it doesn't, I'll modify the Vagrantfile so that it actually builds from source and add that to the repo too.

**Comment by TerrorJack** [comment]
Thanks for making a repro for this issue!

Speaking of docker images, the `Dockerfile` on the `master` branch is unfortunately out of date; we decided to drop the image building on CI due to time limit. I recommend pulling `terrorjack/asterius` directly for now; it's built on a standalone aws instance and has ~2k stackage lts packages built-in which makes it convenient to test stuff. The `Dockerfile`s to make the up to date images exist at the `wip-develop` branch.

**Comment by Hazelfire** [comment]
I'd be very curious to know if that one reproduces the issue for you now... Because halway through testing other variants that worked, the original configuration started working again! It's still an issue within my own project though, I'll see if I can get a better replication if that fails to do so.

**Comment by Hazelfire** [comment]
Ok, I've worked out what the issue is, and I'll slap myself for not finding this earlier. 

I was using a cabal config that used the `-threaded` option. I don't know why it was showing up with other ahc-link'd programs, probably cache. Removing this option fixes the issue.

I don't think that many other people will be making this mistake. But if they do they'll have this issue for reference.

Closing this issue.
